lib/particle/product.rb

```diff
@@ -0,0 +1,77 @@
+require 'particle/model'
+
+module Particle
+
+  # Domain model for one Particle device
+  class Product < Model
+    ID_REGEX = /^\d{1,5}$/
+
+    def initialize(client, attributes)
+      super(client, attributes)
+
+      attributes = attributes.to_s if attributes.is_a?(Integer)
+
+      if attributes.is_a? String
+        if attributes =~ ID_REGEX
+          @attributes = { id: attributes }
+        else
+          @attributes = { slug: attributes }
+        end
+      else
+        # Listing all devices returns partial attributes so check if the
+        # device was fully loaded or not
+        @fully_loaded = true if attributes.key?(:name)
+      end
+    end
+
+    # NOTE: the key :requires_activation_codes is documented (as of 2018/05/19)
```

Comment:
requires_activation_codes is deprecated. You can delete the comment and ignore it.